spec/requests/partners_requests_spec.rb

```diff
@@ -124,4 +124,75 @@
       expect(response).to have_http_status(:found)
     end
   end
+
+  describe "PUT #deactivate" do
+    let(:partner) { create(:partner, organization: @organization) }
+
+    context "when the partner successfully deactivates" do
+      it 'performs a PUT request' do
+        expect(partner.status).not_to eq("deactivated")
+        response = double("Response", value: Net::HTTPSuccess)
+        allow(DiaperPartnerClient).to receive(:put).and_return(response)
+        put deactivate_partner_path(default_params.merge(id: partner.id))
+
+        expect(partner.reload.status).to eq("deactivated")
+        expect(response).to redirect_to(partners_path)
+        expect(flash[:notice]).to eq("#{partner.name} successfully deactivated!")
+      end
```

Comment:
I think this is in the right direction! I'd add a spec to handle the other case when the deactivation fails.
```
context 'when the partner does not successfully deactivate' do
    before do
        # Setup the stubs to force a failed response.
    end

    it 'should <do these things' do.
      put deactivate_partner_path(default_params.merge(id: partner.id))
      # do the assertions here.
    end
end
```